stdlib/private/SwiftPrivateLibcExtras/Subprocess.swift

```diff
@@ -64,12 +67,38 @@ func posixPipe() -> (readFD: CInt, writeFD: CInt) {
 /// stderr.
 public func spawnChild(args: [String])
   -> (pid: pid_t, stdinFD: CInt, stdoutFD: CInt, stderrFD: CInt) {
+  let childStdout = posixPipe()
+  let childStdin = posixPipe()
+  let childStderr = posixPipe()
+
+#if os(Android)
+  // posix_spawn isn't available on Android. Instead, we fork and exec.
+  let pid = fork()
+  if pid < 0 {
+    preconditionFailure("fork() failed")
+  } else if pid == 0 {
+    // pid of 0 means we are now in the child process.
+    // Capture the output and execute the program.
+    dup2(childStdout.writeFD, STDOUT_FILENO)
+    dup2(childStdin.readFD, STDIN_FILENO)
+    dup2(childStderr.writeFD, STDERR_FILENO)
+
+    var result: Int32 = 0
+    withArrayOfCStrings([Process.arguments[0]] + args) {
+      result = execve(Process.arguments[0], $0, _getEnviron())
+    }
+
+    if result != 0 {
+      preconditionFailure("execve() failed. errno: \(errno)")
+    }
+    _exit(126)
```

Comment:
Thank you @weissi, your explanation was incredibly helpful! Thanks for so clearly explaining the problem, it made writing the code much easier. I have a working implementation that I will amend to this commit soon; I'm running the Android test suite from #1714 to confirm it all still works (looks good so far!).

(For those that are interested, there are seven failing tests when the stdlib is compiled for Android. Most are related to assumptions in the test expectations themselves.)